### ISSUE TITLE
worker status idle after enqueuing task

### ISSUE THREAD

**Issue by asabya** [plexsysio/taskmanager#3]
```
        started, err := taskManager.Go(task)
	if err != nil {
		return err
	}
	<-started
	fmt.Println(taskManager.Status())
	<-time.After(time.Millisecond * 100)
	fmt.Println(taskManager.Status())
```

running the avobe code results the follwoing output
```
map[1:{idle waiting}]
map[1:{6a50032864056992563cee7e31b3323bd25ac34c157f658d02b32a59e241f7f3 running}]
```

Even after waiting on the channel `Status()` says worker is idle. Only after waiting some time we get the task in the status

**Comment by aloknerurkar** [comment]
So the key thing here is that the channel returned, signals that the task is `scheduled` and not `started`. This means a worker was found to take up the task. The caller can return knowing for sure that a worker was assigned.

The status is updated by the worker when he starts `Executing` the task. There will be some gap here depending on the scheduling of the goroutines by the CPU. The worker routine needs to get scheduled and dequeue the task. In your case the `<-time.After()` gives up the CPU and the worker can update the status, which is why you would see it updated immediately after.

TLDR; this is expected behavior for now. I guess it could be a little misleading but then is there a hard requirement for `Status` to return info immediately once you start the task?

One workaround here is to manually create a channel and close it in `Execute` to signal execution has started. This can be done completely independent of the taskmanager.

**Comment by asabya** [comment]
Thanks for explaining. It was just a test that was failing for this. Not a hard requirement.
